pkg/discoveredpolicy/controllers/policy_watchers.go

```diff
@@ -0,0 +1,188 @@
+	pw.KspInformer, err = pw.CreateInformer(kspResource)
+	if err != nil {
+		return nil, err
+	}
+	pw.CiliumInformer, err = pw.CreateInformer(ciliumPolicyResource)
+	if err != nil {
+		return nil, err
+	}
+	pw.NetworkPolicyInformer, err = pw.CreateInformer(networkPolicyResource)
+	if err != nil {
+		return nil, err
+	}
+	return pw, nil
+}
+
+func initDynamicClient() (*dynamic.DynamicClient, error) {
+	client, err := dynamic.NewForConfig(ctrl.GetConfigOrDie())
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// CreateInformer func
+func (p *PolicyWatcher) CreateInformer(gvr schema.GroupVersionResource) (cache.SharedIndexInformer, error) {
+	// Create an informer factory using the dynamic client.
+	informerFactory := dynamicinformer.NewDynamicSharedInformerFactory(p.Client, time.Second*30)
+
+	// informers
+	Informer := informerFactory.ForResource(gvr).Informer()
+
+	// event handlers for kspInformer
+	if _, err := Informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			// nothing to be done
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			// nothing to be done
+		},
+		DeleteFunc: func(obj interface{}) {
+			// update dsp policy status
+			pol := obj.(*unstructured.Unstructured)
+			dsp, err := p.Client.
+				Resource(dspResource).
+				Namespace(pol.GetNamespace()).
+				Get(context.TODO(), pol.GetName(), metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					// nothing to be done here
+				}
+				p.Log.Error(err, "Unable to get discovered policy resource")
+				return
+			}
+			// update dsp policy status to inactive
+			patch := []byte(`{"spec": {"status": "Inactive"}}`)
+			_, err = p.Client.Resource(dspResource).
+				Namespace(dsp.GetNamespace()).
+				Patch(context.TODO(), dsp.GetName(), patchTypes.MergePatchType, patch, metav1.PatchOptions{})
+			if err != nil {
+				p.Log.Error(err, "Unable to revert discovered policy status back to inactive")
+				return
+			}
+
+		},
+	}); err != nil {
+		return nil, err
+	}
+
+	return Informer, nil
+
+}
+
+// StartWatchers func
+func (p *PolicyWatcher) StartWatchers(stopCh <-chan struct{}) error {
+	if IsAPIResourceAvailable(kspResource) {
+		go p.KspInformer.Run(stopCh)
+		// Wait for the informer to sync.
+		if ok := cache.WaitForCacheSync(stopCh, p.KspInformer.HasSynced); !ok {
+			return fmt.Errorf("Failed to wait for cache sync")
+		}
+	}
+
+	if IsAPIResourceAvailable(ciliumPolicyResource) {
+		go p.CiliumInformer.Run(stopCh)
+		// Wait for the informer to sync.
+		if ok := cache.WaitForCacheSync(stopCh, p.CiliumInformer.HasSynced); !ok {
+			return fmt.Errorf("Failed to wait for cache sync")
+		}
+	}
+
+	if IsAPIResourceAvailable(networkPolicyResource) {
+		go p.NetworkPolicyInformer.Run(stopCh)
+		// Wait for the informer to sync.
+		if ok := cache.WaitForCacheSync(stopCh, p.NetworkPolicyInformer.HasSynced); !ok {
+			return fmt.Errorf("Failed to wait for cache sync")
+		}
+	}
```

Comment:
Can we create a single function for abstration?

Reply:
Please address this too

Reply:
imo we should be good here, it's already abstracted to an extent.